fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeUtils.java

```diff
@@ -148,7 +158,22 @@ public static void injectBiomesIntoClimate(LayerRandomnessSource random, int[] v
 		}
 	}
 
-	public static void recreateChunkGenerators(GeneratorOptions generatorOptions) {
-		((DimensionOptionsAccessor) (Object) generatorOptions.getDimensionMap().get(DimensionOptions.NETHER)).setChunkGenerator(DimensionTypeAccessor.createNetherGenerator(generatorOptions.getSeed()));
+	public static int getRawId(RegistryKey<Biome> key) {
+		return BuiltinRegistries.BIOME.getRawId(BuiltinRegistries.BIOME.method_31140(key));
+	}
+
+	/**
+	 * Makes sure that the given registry key is mapped in {@link Biomes}. This mapping may be absent
+	 * if mods register their biomes only in {@link BuiltinRegistries#BIOME}, and not using the
+	 * private method in {@link Biomes}.
+	 */
+	public static void ensureIdMapping(RegistryKey<Biome> biomeKey) {
```

Comment:
This validation doesn't seem great to me. How does this interact with registering to `DynamicRegistryManager.BUILTIN` or using json biomes?
I'd suggest dropping this validation for the initial PR.
This seems like it could cause issues, and that's the last thing we want validation to do.

Reply:
It doesn't interact with it. This API will be used by modders in their Mod initializers, when no dynamic registry manager exists. The only way to ensure only properly registered Biomes are used is to validate against the BuiltInRegistries, and follow the same general approach Vanilla does.